### ISSUE TITLE
PUT methods for objects

### ISSUE THREAD

**Issue by almartin82** [cloudyr/aws.s3#7]
would love to be helpful here - was able to get `getbucket()` to run, but after digging around in the code a bit, I noticed that the header construction was a lot different in `getbucket()` than in any of the other methods (eg all the stuff for objects).  I rolled my own function cribbing off of `getbucket` and was able to write to my s3 bucket.

anything pointers/suggestions about what I should be doing differently?  I see a lot of `...` arguments in [object.r](https://github.com/cloudyr/aws.s3/blob/08a02e66ba85b9dc1437de94f9dde8796ec25a50/R/OBJECT.r) -- should I be building my headers using some constuctor, then passing that to `putobject`, etc?  

will be happy to write up whatever I learn and submit back as a vignette / documentation / whatever

**didn't work**
```
> aws.s3::putobject('my_bucket', 1)
Error: is.character(headers) is not TRUE
Called from: request(headers = c(..., .headers))
```

**works**
```
putobject_alm <- function(
  bucket, 
  object,                        
  prefix, 
  delimiter,
  max,
  marker, 
  ...
) {
  if(inherits(bucket, "s3_bucket"))
      bucket <- bucket$Name
  h <- list()
  if(!missing(prefix))
      h$prefix <- prefix
  if(!missing(delimiter))
      h$delimiter <- delimiter
  if(!missing(max))
      h$"max-keys" <- max
  if(!missing(marker))
      h$marker <- marker
  h$`x-amz-content-sha256` <- ""
  r <- s3HTTP("PUT", paste0("https://",bucket,".s3.amazonaws.com/", object), 
    headers = h, ...
  )
  if(inherits(r, "aws_error")) {
      return(r)
  } else {
      structure(r, class = "s3_object")
  }
}

putobject_alm('my_bucket', 1)
```

**Comment by almartin82** [comment]
the header issue above was fixed by [212e598](https://github.com/cloudyr/aws.s3/commit/212e598a4e0381bb0a77f39f7ff9ddc23e42ec35).  will revist the PUT stuff generally after tests are in place for the GET methods.

**Comment by leeper** [comment]
@almartin82 Make sure to add yourself as a package contributor to `DESCRIPTION` at some point.
